### PR TITLE
Added border outline to project icons

### PR DESCRIPTION
**Pull request by TurnbulA**

### CODE REVIEW

File: app/js/toDoToggle.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ const openProject = () => {
       projectIconBody.classList.add("activeProject");
       projectToDo.classList.add("activeList");
       e.target.classList.add("iconActive");
+      buttonClick = e.target;
```
Comment (mali891) [inline app/js/toDoToggle.js]: This naming is off

File: app/js/buildProjects.js [hunk 2]
```diff
@@ -61,10 +61,12 @@ const renderIcon = () => {
...
       const words = getTitleInitials(item.title);
       return `<li class="c-projects__list--item project-Icon" data-id=${item.dataId}>
+        <div class="c-projects--border u-gradient-bg--${item.colour}--border" data-id=${item.dataId}></div>
             <div class="c-projects--icon u-gradient-bg--${item.colour} project-Icon" data-id=${item.dataId}>
               <h1 class="c-projects--initals" data-id=${item.dataId}>${words}</h1>
             </div>
-            <h2>${item.title}</h2>
+          
+          <h2>${item.title}</h2>
         </li>`;
```
Comment (mali891) [inline app/js/buildProjects.js]: All of these things can be destructured out of `items`

File: app/scss/projectIcons.scss [hunk 3]
```diff
@@ -33,4 +33,16 @@
   &--initals {
     color: $blackSqueeze;
   }
+  &--border {
+    width: 15rem;
+    height: 15rem;
+    position: relative;
+    display: none;
+    content: "";
+    margin-top: -3.5rem;
```
Comment (mali891) [inline app/scss/projectIcons.scss]: What's going on here?

File: app/scss/projectIcons.scss [hunk 4]
```diff
@@ -33,4 +33,16 @@
...
+    z-index: -1;
```
Comment (mali891) [inline app/scss/projectIcons.scss]: Tell me more

File: app/js/buildProjects.js [hunk 5]
```diff
@@ -61,10 +61,12 @@ const renderIcon = () => {
     .map(item => {
```
Comment (mali891) [inline app/js/buildProjects.js]: instead of `(item) => { }`, this can be `({ colour, dataId, title }) => { }`

Comment (mali891) [reply app/js/buildProjects.js]: ?

File: app/js/toDoToggle.js [hunk 6]
```diff
@@ -6,24 +6,30 @@ const openProject = () => {
         ".c-container--project-body"
       );
       projectTarget = e.target.getAttribute("data-id");
+      iconBorders = [...document.querySelectorAll(".c-projects--icon--border")];
```
Comment (mali891) [inline app/js/toDoToggle.js]: Why are there two modifiers in this?

Comment (TurnbulA) [reply app/js/toDoToggle.js]: Coz for reasons

File: app/js/toDoToggle.js [hunk 7]
```diff
@@ -6,24 +6,30 @@ const openProject = () => {
...
+      iconBorders.forEach(iconBorder => {
+        if (projectTarget === iconBorder.getAttribute("data-id")) {
+          iconBorder.classList.add("borderActive");
```
Comment (mali891) [inline app/js/toDoToggle.js]: surely `active` should be a modifier here?

File: app/js/toDoToggle.js [hunk 8]
```diff
@@ -6,24 +6,30 @@ const openProject = () => {
...
 };
 openProject();
+
 const closeProject = () => {
   const projectIconBody = document.querySelector(".c-container--project-body");
   const projectToDo = document.querySelector(".c-container--project-list");
   const activeProjects = [...document.querySelectorAll(".iconActive")];
   projectIconBody.classList.remove("activeProject");
   projectToDo.classList.remove("activeList");
+  document.querySelector(".borderActive").classList.remove("borderActive");
```
Comment (mali891) [inline app/js/toDoToggle.js]: And here

File: app/scss/utilities.scss [hunk 9]
```diff
@@ -1,83 +1,106 @@
...
-    background-image: linear-gradient(to top right, $pink2, $pink1);
-    &:hover {
-      box-shadow: 0 0 2rem rgba(248, 59, 200, 1);
-    }
-    &.isActive {
-      box-shadow: 0 0 2rem rgba(248, 59, 200, 1);
+.u-gradient {
+  &-border {
+    &--pink {
+      &.borderActive {
```
Comment (mali891) [inline app/scss/utilities.scss]: Why do they need this `borderActive` class?

File: app/scss/utilities.scss [hunk 10]
```diff
@@ -1,83 +1,106 @@
...
-      border-radius: 3rem;
+    &--orange {
+      background-image: linear-gradient(to top right, $orange1, $orange2);
+      &:hover {
+        box-shadow: 0 0 2rem rgba(248, 190, 115, 1);
+      }
+      &.isActive {
+        box-shadow: 0 0 2rem rgba(248, 190, 115, 1);
+      }
+      &--border {
```
Comment (mali891) [inline app/scss/utilities.scss]: `--border` is not a modifier

Comment (TurnbulA) [reply app/scss/utilities.scss]: That's from a previous version
I forgot to get rid of it

File: app/js/toDoToggle.js [hunk 11]
```diff
@@ -53,5 +52,20 @@ const closeProject = () => {
 document
   .querySelector(".c-projects-to-do__close--cross")
   .addEventListener("click", () => {
-    closeProject();
+    if (document.querySelector(".borderActive")) {
```
Comment (mali891) [inline app/js/toDoToggle.js]: Why `borderActive`

File: app/js/toDoToggle.js [hunk 12]
```diff
@@ -3,20 +3,20 @@ const openProject = () => {
...
       const projectIconBody = document.querySelector(
-        ".c-container--project-body"
+        ".c-container__project-body"
       );
       projectTarget = e.target.getAttribute("data-id");
       iconBorders = [...document.querySelectorAll(".c-projects__icon--border")];
       iconBorders.forEach(iconBorder => {
         if (projectTarget === iconBorder.getAttribute("data-id")) {
-          iconBorder.classList.add("borderActive");
+          iconBorder.classList.add("c-projects__icon--border--active");
```
Comment (mali891) [inline app/js/toDoToggle.js]: Why two modifiers?

File: app/js/toDoToggle.js [hunk 13]
```diff
@@ -3,20 +3,20 @@ const openProject = () => {
...
         }
       });
       renderToDoList(projectTarget);
-      const projectToDo = document.querySelector(".c-container--project-list");
-      projectIconBody.classList.add("activeProject");
-      projectToDo.classList.add("activeList");
-      e.target.classList.add("iconActive");
+      const projectToDo = document.querySelector(".c-container__project-list");
+      projectIconBody.classList.add("c-container__project-body--left-justify");
```
Comment (mali891) [inline app/js/toDoToggle.js]: The width thing is related here, but fix this in the next PR

Comment (mali891) [inline app/js/toDoToggle.js]: Does `project-body` need to be an element on `c-container`? Can't it be its own thing?

File: app/js/toDoToggle.js [hunk 14]
```diff
@@ -3,20 +3,20 @@ const openProject = () => {
...
+      projectToDo.classList.add("c-container__project-list--active");
+      e.target.classList.add("c-projects__icon--active");
```
Comment (mali891) [inline app/js/toDoToggle.js]: Does `icon` need to be an element on `c-projects`? Can't it be its own thing?

File: app/js/toDoToggle.js [hunk 15]
```diff
@@ -27,22 +27,24 @@ const closeProject = () => {
...
-          .querySelector(".activeProject")
-          .classList.remove("activeProject");
+          .querySelector(".c-container__project-body--left-justify")
+          .classList.remove("c-container__project-body--left-justify");
       }
-      if (document.querySelector(".activeList")) {
-        document.querySelector(".activeList").classList.remove("activeList");
+      if (document.querySelector(".c-container__project-list--active")) {
+        document
+          .querySelector(".c-container__project-list--active")
```
Comment (mali891) [inline app/js/toDoToggle.js]: Save these in a `const`

File: app/js/toDoToggle.js [hunk 16]
```diff
@@ -27,22 +27,24 @@ const closeProject = () => {
...
       activeIcons.forEach(activeIcon => {
-        activeIcon.classList.remove("iconActive");
+        activeIcon.classList.remove("c-projects__icon--active");
       });
-      if (document.querySelector(".activeProject")) {
+      if (document.querySelector(".c-container__project-body--left-justify")) {
         document
-          .querySelector(".activeProject")
-          .classList.remove("activeProject");
+          .querySelector(".c-container__project-body--left-justify")
```
Comment (mali891) [inline app/js/toDoToggle.js]: Save these in a `const`

File: app/js/toDoToggle.js [hunk 17]
```diff
@@ -52,20 +54,24 @@ const closeProject = () => {
 document
   .querySelector(".c-projects-to-do__close--cross")
   .addEventListener("click", () => {
-    if (document.querySelector(".borderActive")) {
-      document.querySelector(".borderActive").classList.remove("borderActive");
+    if (document.querySelector(".c-projects__icon--border--active")) {
+      document
```
Comment (mali891) [inline app/js/toDoToggle.js]: Save these in a `const`

File: app/js/toDoToggle.js [hunk 18]
```diff
@@ -52,20 +54,24 @@ const closeProject = () => {
...
       document
-        .querySelector(".activeProject")
-        .classList.remove("activeProject");
+        .querySelector(".c-container__project-body--left-justify")
+        .classList.remove("c-container__project-body--left-justify");
     }
-    if (document.querySelector(".activeList")) {
-      document.querySelector(".activeList").classList.remove("activeList");
+    if (document.querySelector(".c-container__project-list--active")) {
+      document
```
Comment (mali891) [inline app/js/toDoToggle.js]: Save these in a `const`

File: app/js/toDoToggle.js (hunk 14 above)
Comment (TurnbulA) [reply app/js/toDoToggle.js]: Why when its a project icon?

File: app/js/toDoToggle.js [hunk 19]
```diff
@@ -1,50 +1,48 @@
...
       });
       renderToDoList(projectTarget);
-      const projectToDo = document.querySelector(".c-container__project-list");
-      projectIconBody.classList.add("c-container__project-body--left-justify");
-      projectToDo.classList.add("c-container__project-list--active");
-      e.target.classList.add("c-projects__icon--active");
+      const projectToDo = document.querySelector(".c-project__list");
+      projectIconBody.classList.add("c-project__body--left-justify");
+      projectToDo.classList.add("c-project__list--active");
+      e.target.classList.add("c-projects-icon__active");
```
Comment (mali891) [inline app/js/toDoToggle.js]: `active` is a modifier, because it describes the state of a thing

File: app/js/toDoToggle.js (hunk 17 above)
Comment (mali891) [reply app/js/toDoToggle.js]: helloooo

File: app/js/toDoToggle.js [hunk 20]
```diff
@@ -54,24 +52,24 @@ const closeProject = () => {
...
+        .classList.remove("c-projects-icon__border--active");
     }
-    activeIcons = [...document.querySelectorAll(".c-projects__icon--active")];
+    activeIcons = [...document.querySelectorAll(".c-projects-icon__active")];
     activeIcons.forEach(activeIcon => {
-      activeIcon.classList.remove("c-projects__icon--active");
+      activeIcon.classList.remove("c-projects-icon__active");
     });
-    if (document.querySelector(".c-container__project-body--left-justify")) {
+    if (document.querySelector(".c-project__body--left-justify")) {
```
Comment (mali891) [inline app/js/toDoToggle.js]: `const`

File: app/js/toDoToggle.js [hunk 21]
```diff
@@ -54,24 +52,24 @@ const closeProject = () => {
...
+        .querySelector(".c-projects-icon__border--active")
+        .classList.remove("c-projects-icon__border--active");
     }
-    activeIcons = [...document.querySelectorAll(".c-projects__icon--active")];
+    activeIcons = [...document.querySelectorAll(".c-projects-icon__active")];
     activeIcons.forEach(activeIcon => {
-      activeIcon.classList.remove("c-projects__icon--active");
+      activeIcon.classList.remove("c-projects-icon__active");
     });
-    if (document.querySelector(".c-container__project-body--left-justify")) {
```
Comment (mali891) [inline app/js/toDoToggle.js]: `const`

File: app/js/toDoToggle.js (hunk 18 above)
Comment (mali891) [reply app/js/toDoToggle.js]: `const`

File: app/js/buildProjects.js [hunk 22]
```diff
@@ -60,9 +60,9 @@ const renderIcon = () => {
   const projectIcons = Object.values(projects)
     .map(item => {
       const words = getTitleInitials(item.title);
-      return `<li class="c-projects__list--item project-Icon " data-id=${item.dataId}>
-          <div class="c-projects-icon__border u-gradient-border--${item.colour}" data-id=${item.dataId}>
-              <div class="c-projects-icon u-gradient-bg--${item.colour} project-Icon" data-id=${item.dataId}>
+      return `<li class="c-projects__list--item project-icon " data-id=${item.dataId}>
```
Comment (mali891) [inline app/js/buildProjects.js]: `c-project-icon`?

File: app/js/buildProjects.js [hunk 23]
```diff
@@ -60,9 +60,9 @@ const renderIcon = () => {
...
+          <div class="c-icon__border u-gradient-border--${item.colour}" data-id=${item.dataId}>
+              <div class="c-icon u-gradient-bg--${item.colour} project-icon" data-id=${item.dataId}>
```
Comment (mali891) [inline app/js/buildProjects.js]: `c-project-icon`?

File: app/js/buildProjects.js [hunk 24]
```diff
@@ -58,12 +58,12 @@ const getTitleInitials = title => {
...
-                <h1 class="c-projects--initals"  data-id=${item.dataId}>${words}</h1>
+    .map({colour, dataId, title} => {
+      const words = getTitleInitials(title);
+      return `<li class="c-projects__list--item  " data-id=${dataId}>
+          <div class="c-icon__border u-gradient-border--${colour}" data-id=${dataId}>
+              <div class="c-icon u-gradient-bg--${colour} " data-id=${dataId}>
+                <h1 class="c-projects--initals"  data-id=${dataId}>${words}</h1>
               </div>
             </div>
           <h2>${item.title}</h2>
```
Comment (mali891) [inline app/js/buildProjects.js]: And this guy

**Pull request merged**